src/tactic/ring2.lean

```diff
@@ -39,21 +42,34 @@ def tree.index_of {α} (lt : α → α → Prop) [decidable_rel lt]
   | ordering.gt := pos_num.bit1 <$> tree.index_of t₂
   end
 
+/-- Reflects a tree of expressions into an expression that,
+when evaluated, yields the original tree. -/
```

Comment:
This isn't clear to me. What do you mean the "original tree"? What is the type of the resulting expr?